### PR TITLE
Support `servant-0.20` and GHC 9.6

### PR DESCRIPTION
**Pull request by ocharles**

### CODE REVIEW

**Comment by jhrcek** [comment]
I'll look at this and try to update the package & CI over the next few days.

**Comment by ocharles** [comment]
Great, thank you! :pray:

File: servant-hmac-auth.cabal [hunk 1]
```diff
@@ -74,12 +74,12 @@ library
...
                      , memory >= 0.15 && < 0.19
-                     , mtl ^>= 2.2.2
-                     , servant ^>= 0.18 || ^>= 0.19
-                     , servant-client ^>= 0.18 || ^>= 0.19
-                     , servant-client-core ^>= 0.18 || ^>= 0.19
-                     , servant-server ^>= 0.18 || ^>= 0.19
-                     , transformers ^>= 0.5
+                     , mtl ^>= 2.2.2 || ^>= 2.3
+                     , servant ^>= 0.18 || ^>= 0.19 || ^>= 0.20
+                     , servant-client ^>= 0.20
```
[the remark this answers is not in the document]
Comment (jhrcek) [reply servant-hmac-auth.cabal:79]: I'm gonna make it work with servant-0.19. It should be a matter of 1 sensibly placed CPP

**Comment by jhrcek** [comment]
I'm trying to make it work in https://github.com/Holmusk/servant-hmac-auth/pull/65

**Comment by jhrcek** [comment]
Superseded by https://github.com/Holmusk/servant-hmac-auth/pull/65
I'm gonna upload new version of this package to hackage later today.

**Pull request closed without merging**